### DISCUSSION TITLE
support for vector-effect="non-scaling-stroke"

### DISCUSSION THREAD

**Discussion by P4bloo** [mono/SkiaSharp.Extended#232] [Q&A]
HI Matthew @mattleibow,
is there a chance that vector-effect="non-scaling-stroke" rule can be also supported or considered in the future? 

Thanks a lot,
Pabloo

**Comment by mattleibow** [comment]
When you say support, do you mean you want to read a SVG that is using that? You may want to ask the authors on the parsing side: https://github.com/wieslawsoltes/Svg.Skia

**Reply by P4bloo** [reply]
HI Matthew @mattleibow , I mean the effect it provides.
  
I use:

`var svg = new SkiaSharp.Extended.Svg.SKSvg();
        svg.Load("input.svg");`

to read my svg file: 

`<svg xmlns="http://www.w3.org/2000/svg"  width="400" height="400"  >
  <rect x="50" y="20" width="150" height="75"
  style="fill:none;stroke:red;stroke-width:1" transform="scale(2)" vector-effects="non-scaling-stroke" />
</svg>`

Next, I draw and save it: 

```
using (SkiaSharp.SKBitmap bitmap = new SkiaSharp.SKBitmap((int)svg.CanvasSize.Width, (int)svg.CanvasSize.Height))
        using (SkiaSharp.SKCanvas canvas = new SkiaSharp.SKCanvas(bitmap))
        {
            canvas.DrawPicture(svg.Picture, paint);
            using (SKImage image = SKImage.FromBitmap(bitmap))
            using (SKData data = image.Encode(SKEncodedImageFormat.Png, 80))
            using (var stream = File.OpenWrite("output.png"))
            {
                data.SaveTo(stream);
            }
        }

```
The rectangle is scaled and drawn, but it seems that the vector effect is not being taken into account, and the stroke-width is adjusted accordingly. I would like to scale the object and keep stroke-width unchanged.
